src/templates/plugin.main.js

```diff
@@ -174,11 +175,31 @@ export default async (context) => {
       (detectBrowserLanguage && doDetectBrowserLanguage(route)) ||
       getLocaleFromRoute(route) || app.i18n.locale || app.i18n.defaultLocale || ''
 
-    await app.i18n.setLocale(finalLocale)
+    if (skipSettingLocaleOnNavigate) {
+      app.i18n.__pendingLocale = finalLocale
+      app.i18n.__pendingLocalePromise = new Promise(resolve => {
+        app.i18n.__resolvePendingLocalePromise = resolve
+      })
+    } else {
+      await app.i18n.setLocale(finalLocale)
+    }
 
     return [null, null]
   }
 
+  const setPendingLocale = async () => {
+    if (!app.i18n.__pendingLocale) {
+      return
+    }
+    await app.i18n.setLocale(app.i18n.__pendingLocale)
+    app.i18n.__resolvePendingLocalePromise()
+    app.i18n.__pendingLocale = null
+  }
+
+  const waitForPendingLocale = async () => {
+    return !app.i18n.__pendingLocale || (await app.i18n.__pendingLocalePromise)
```

Comment:
Would prefer this as it would be more correct from the perspective of the types (I'm planning to enable more strict type-checking at some point):
```suggestion
    if (app.i18n.__pendingLocale) {
      await app.i18n.__pendingLocalePromise
    }
```